core/src/Streamly/Data/Array/Unboxed/Mut.hs

```diff
@@ -19,7 +19,8 @@ module Streamly.Data.Array.Unboxed.Mut
     -- ** Construction
 
     -- Uninitialized Arrays
-    , newArray
+    , newPinned
+    , newUnpinned
```

Comment:
This should probably be the default. So that without thinking about pinned/unpinned you always create an unpinned array unless you specifically require pinned. Usually pinning will be handled transparently by IO operations and not required for normal use cases.